### PR TITLE
Fix:the rounded corners of the folded file are not displayed correctly

### LINKED ISSUE #29933: The corners of the folded review files in pull request conversation page are not right.
image

### PR DESCRIPTION
**Pull request by HEREYUA**
Fix:    [#29933](https://github.com/go-gitea/gitea/issues/29933)

**Before**

image

**After**

image

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by silverwind** [comment]
How does it look in expanded state?

**Comment by HEREYUA** [comment]
It appears to display normally in the expanded state
image

**Comment by silverwind** [comment]
Looks okay, but I will check in detail later if this can be improved further so it only targets the collapsed state.

**Review by delvh — APPROVED** [review]

**Review by silverwind — CHANGES_REQUESTED** [review]
Block until I have investigated.

**Comment by silverwind** [comment]
I pushed another alternative fix:

```css
.ui.segments:not(.horizontal) > .segment:has(~ .tw-hidden) {
  border-radius: 0.28571429rem;
}
```

This selector means "any segment that is followed by one or more hidden elements". In a situation with no hidden elements, `.ui.segments:not(.horizontal) > .segment:last-child` would match but with hidden elements, the method breaks because then the element is no longer the last child.

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
@wxiaoguang fyi, above is another case where selecting for `.tw-hidden` is the only way to fix an issue 😉

**Review by lunny — APPROVED** [review]

**Review by wxiaoguang — COMMENTED** [review]
I think it is abused. I will take a look.

**Comment by silverwind** [comment]
The only cleaner solution that I see it to remove collapsed elements from the DOM. It may be possible because I think this toggling is happening in our JS. Another solution may be to work something out with `<details>` element.

File: web_src/css/modules/segment.css [hunk 1]
```diff
@@ -69,7 +69,8 @@
   border-radius: 0 0 0.28571429rem 0.28571429rem;
 }
 
-.ui.segments:not(.horizontal) > .segment:only-child {
+.ui.segments:not(.horizontal) > .segment:only-child,
+.ui.segments:not(.horizontal) > .segment:has(~ .tw-hidden) {
```
Comment (silverwind) [inline web_src/css/modules/segment.css]:
```suggestion
.ui.segments:not(.horizontal) > .segment:has(~ .tw-hidden) { /* workaround css issue with :last-child ignoring hidden elements */
```

File: web_src/css/modules/segment.css [hunk 2]
```diff
@@ -69,7 +69,8 @@
   border-radius: 0 0 0.28571429rem 0.28571429rem;
 }
 
-.ui.segments:not(.horizontal) > .segment:only-child {
+.ui.segments:not(.horizontal) > .segment:only-child,
+.ui.segments:not(.horizontal) > .segment:has(~ .tw-hidden) { /* workaround css issue with :last-child ignoring hidden elements */
```
Comment (silverwind) [inline web_src/css/modules/segment.css]:
```suggestion
.ui.segments:not(.horizontal) > .segment:has(~ .tw-hidden) { /* workaround issue with :last-child ignoring hidden elements */
```

**Pull request merged**